src/FlightMap/Widgets/AltitudeWidget.qml

```diff
@@ -0,0 +1,184 @@
+/****************************************************************************
+ *
+ * (c) 2009-2020 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ *
+ * QGroundControl is licensed according to the terms in the file
+ * COPYING.md in the root of the source code directory.
+ *
+ ****************************************************************************/
+
+import QtQuick  2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import QGroundControl               1.0
+import QGroundControl.ScreenTools   1.0
+import QGroundControl.Controls      1.0
+import QGroundControl.Palette       1.0
+
+Rectangle {
+    id:         root
+    radius:     _toolsMargin / 2
+    color:      qgcPal.window
+    opacity:    0.9
+    clip:       true
+    height:     300
+
+    property var  activeVehicle:     QGroundControl.multiVehicleManager.activeVehicle
+    property real vehicleAltitude:   activeVehicle ? activeVehicle.altitudeAMSL.rawValue : 0
+    property var  missionController: missionController
+    property real distanceToNext:    _calculateDistanceToNext()
+    property real centerAltitude:   _calculateCenterAltitude()
+    property real margins:          ScreenTools.defaultFontPixelWidth / 2
+    property real scaleHeight:      height - (margins * 2) - ScreenTools.defaultFontPixelHeight
+    property real centerY:          (height - ScreenTools.defaultFontPixelHeight) / 2
+    property real availableHeight:  300 // Default value, will be overridden by parent
+
+    property real metersBetweenLines:         10
+    property real minVisibleRangeInMeters:    40
+    property real visibleRange:               40
+    property real pixelsPerMeter:             scaleHeight / visibleRange
+
+    property bool vehicleFlying: activeVehicle ? activeVehicle.flying : false
+
+    QGCPalette { id: qgcPal }
+
+    function _calculateDistanceToNext() {
```

Comment:
Is it possible to do this without accessing global scope? There is less room for fuckup with local scope